src/firmware/app/control/trajectory_planner_impl.h

```diff
@@ -0,0 +1,275 @@
+ * constant parameterization. This array is of length num_elements-1
+ *
+ * Node1  |--segment length---| Node2
+ *   ->   *-------------------*  <-
+ */
+void app_trajectory_planner_generate1dSegmentNodesAndLengths_impl(
+    float t_start, float t_end, Polynomial1dOrder3_t path_1d, unsigned int num_elements,
+    float *node_values, float *segment_lengths);
+
+
+/**
+ * Function generates the length of segments in a 2d polynomial. It works as an wrapper
+ * for the euclidean distance and values of 2d polynomials.
+ *
+ * @pre All arrays must be pre-allocated up to at least
+ * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ *
+ * @param t_start [in] The starting value of polynomial parameterization
+ *
+ * @param t_end [in] The final value of polynomial parameterization
+ *
+ * @param path_2d [in] The polynomial representing the trajectory in space
+ *
+ * @param num_elements [in] The number of nodes(elements) of the generated trajectory
+ *
+ * @param x_values [out] The x value of each trajectory node. This array is of length
+ * num_elements.
+ *
+ * @param y_values [out] The y value of each trajectory node. THis array is of length
+ * num_elements
+ *
+ * @param segment_lengths [out] The length of each trajectory segment (euclidean of the XY
+ * 2d polynomial). This array is of length num_elements-1
+ */
+void app_trajectory_planner_generate2dSegmentNodesAndLengths_impl(
+    float t_start, float t_end, Polynomial2dOrder3_t path_2d, unsigned int num_elements,
+    float *x_values, float *y_values, float *segment_lengths);
+
+/**
+ * Function that modifies an existing speed profile to be backwards continuous. This means
+ * that it is possible to reach every speed node within the limits of deceleration
+ *
+ * @pre All arrays are pre-allocated up to at least TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ *
+ * @param segment_lengths [in] The length of each segment in the trajectory.  Units of
+ * parameters must be in consistent magnitudes, ie: distances in mm and speed in mm/s.
+ *
+ * @param max_allowable_acceleration [in] The maximum acceleration that can occur anywhere
+ * on the trajectory
+ *
+ * @param initial_speed [in] The initial speed at the beginning of the trajectory.  Units
+ * of parameters must be in consistent magnitudes, ie: distances in mm and acceleration in
+ * mm/s.
+ *
+ * @param num_segments [in] The number of nodes(elements) in the trajectory
+ *
+ * @param speeds [in/out] The existing speed profile to be modified in-place to be
+ * backwards continuous. This array is of length num_elements. The units of output are
+ * consistent with the input parameters.
+ *
+ * @return A status indicating whether or not modification was successful
+ */
+TrajectoryPlannerGenerationStatus_t
+app_trajectory_planner_modifySpeedsToBeBackwardsContinuous_impl(
+    unsigned int num_segments, float *segment_lengths, float max_allowable_acceleration,
+    float initial_speed, float *speeds);
```

Comment:
Looks like the jdoc doesn't match the arg order here (`num_segments` is out of position).

Reply:
Fixed 👍

Reply:
did this get pushed? still seeems out of order

Reply:
Should be fixed now 👍

Reply:
Looks like `initial_speed` is still out of position here?